filters/auth/oauth.go

```diff
@@ -173,6 +235,37 @@ func jsonGet(url *url.URL, auth string, doc interface{}) error {
 	return d.Decode(doc)
 }
 
+// jsonPost requests url with access token in the body, if auth was given and writes into doc.
+func jsonPost(u *url.URL, auth string, doc *tokenIntrospectionInfo) error {
+	if auth == "" {
+		return fmt.Errorf("invalid without auth")
```

Comment:
this error currently doesn't happen because the token is checked before. If we want to keep this error here, then a better message would be required. E.g: fmt.Errorf("missing auth token for %v", u)